cgi-bin/DW/API/Parameter.pm

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/perl
+#
+# DW::API::Parameter
+#
+# Defines Parameter objects and provides helper functions
+# for use in DW::Controller::API::REST resources.
+#
+# Authors:
+#      Ruth Hatch <ruth.s.hatch@gmail.com>
+#
+# Copyright (c) 2017 by Dreamwidth Studios, LLC.
+#
+# This program is free software; you may redistribute it and/or modify it under
+# the same terms as Perl itself. For a copy of the license, please reference
+# 'perldoc perlartistic' or 'perldoc perlgpl'.
+#
+
+package DW::API::Parameter;
+
+use strict;
+use warnings;
+use JSON;
+
+use Carp qw(croak);
+
+my @ATTRIBUTES = qw(name type in desc);
+my @LOCATIONS = qw(path formData body header query);
+
+# Usage: define_parameter ( \%args ) where arg keys are
+# name, desc, in, type, or required. Creates and returns
+# a new parameter object for use in DW::Controller::API::REST
+# resource definitions.
+
+sub define_parameter {
+    my $args = $_[0];
+    my $parameter = {
+        name => $args->{name},
+        desc => $args->{description},
+        in => $args->{in},
+        type => $args->{type},
+        required => $args->{required}
+    };
+    return bless $parameter;
```

Comment:
I left a comment on the last diff, can we update these to use the preferred style like in DW::Media etc? Basically something like,

```
sub define_parameter {
    my ( $class, $args ) = @_;
    my $parameter = { ... };
    return bless $parameter, $class;
}
```

The reason for this is that it makes it explicit that you're blessing into a certain class and also the calling convention (using the `->` notation) is more idiomatic.

Reply:
How would calling the new variant work? That was the part that really confused me - I didn't know how to make it actually work and not just break horribly.